pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2227,3 +2227,21 @@ def test_categorical_non_unique_monotonic(n_categories):
         index=left_index,
     )
     tm.assert_frame_equal(expected, result)
+
+
+@pytest.mark.parametrize(
+    ("kwargs", "args"),
+    [
+        ({"left_on": ["b"], "right_index": True}, 1),
+        ({"left_index": True, "right_on": ["b"]}, -1),
+    ],
+)
+def test_merge_empty_right_index_left_on(kwargs, args):
+    # GH 33814
+    df1 = pd.DataFrame({"a": [1], "b": [2]}).set_index(["a", "b"])
+    df2 = pd.DataFrame({"b": [1]}).set_index(["b"])
+    result = pd.merge(*[df1, df2][::args], **kwargs)
+    expected = pd.DataFrame(
+        index=MultiIndex(levels=[[int()], [int()]], codes=[[], []], names=["a", "b"])
```

Comment:
this index construction is very odd, why is this not just `levels=[[], []]` ?

Reply:
The dtype was wrong in this case. But creating it via DataFrame and set_index works and is much better Imo.